sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/PaginatedDocuments.java

```diff
@@ -118,26 +145,14 @@ public boolean hasNextPage() {
      */
     public AppCenterFuture<Page<T>> getNextPage() {
         final DefaultAppCenterFuture<Page<T>> result = new DefaultAppCenterFuture<>();
-        if (hasNextPage()) {
-            CosmosDb.callCosmosDbListApi(
-                    mTokenResult,
-                    mContinuationToken,
-                    mHttpClient,
-                    new ServiceCallback() {
-
-                        @Override
-                        public void onCallSucceeded(String payload, Map<String, String> headers) {
-                            Page<T> page = Utils.parseDocuments(payload, mDocumentType);
-                            mCurrentPage = page;
-                            mContinuationToken = headers.get(Constants.CONTINUATION_TOKEN_HEADER);
-                            result.complete(page);
-                        }
-
-                        @Override
-                        public void onCallFailed(Exception e) {
-                            result.complete(new Page<T>(e));
-                        }
-                    });
+        if (hasNextPage() && mNextPageDelegate != null) {
+            DefaultAppCenterFuture<PaginatedDocuments<T>> paginatedResult = new DefaultAppCenterFuture<>();
+            mNextPageDelegate.loadNextPage(mTokenResult, paginatedResult, mReadOptions, mDocumentType, mContinuationToken);
+            PaginatedDocuments<T> docs = paginatedResult.get();
```

Comment:
we are making an asynchronous API synchronous with this call.
We need to modify this to use thenAccept.

Also it might deadlock if called on UI thread on a real app (not on tests). That will also be fixed by making the call asynchronous.

It was async before so this is a regression.